Userland/Games/GameOfLife/BoardWidget.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2021, Andres Crucitti <dasc495@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+#include "BoardWidget.h"
+#include <LibGUI/Painter.h>
+
+BoardWidget::BoardWidget(size_t columns, size_t rows)
+{
+    m_timer = add<Core::Timer>();
+    m_timer->stop();
+    m_timer->on_timeout = [this] {
+        run_generation();
+    };
+    m_timer->set_interval(m_running_timer_interval);
+
+    update_board(columns, rows);
+}
+
+void BoardWidget::run_generation()
+{
+    m_board->run_generation();
+    update();
+    if (m_board->is_stalled()) {
+        if (on_stall)
+            on_stall();
+        update();
+    };
+}
+
+void BoardWidget::update_board(size_t columns, size_t rows)
+{
+    set_running(false);
+
+    m_last_cell_toggled = columns * rows;
+
+    if (m_board) {
+        if (columns == m_board->columns() && rows == m_board->rows()) {
+            return;
+        }
+    }
+
+    delete m_board;
+    m_board = new Board(columns, rows);
```

Comment:
Please don't use naked `new` and `delete`. Instead, use `make<Board>(...)` to create a new Board and store it in an `OwnPtr<Board>`.